.github/workflows/ci-pipeline.yml

```diff
@@ -1,18 +1,42 @@
 name: Continuous Integration
-on: 
+on:
   push:
     paths-ignore:
-    - 'README.md'
+      - "README.md"
   pull_request:
     paths-ignore:
-    - 'README.md'
+      - "README.md"
 jobs:
   build:
     name: Build and test
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - name: Build docker image
-      run: docker build --target test --tag my-test-image .
-    - name: Run tests
-      run: docker run my-test-image
\ No newline at end of file
+      - uses: actions/checkout@v2
+      - name: Build docker image
+        run: docker build --target test --tag my-test-image .
+      - name: Run tests
+        run: docker run my-test-image
+
+  deploy:
+    name: Push to Docker Hub
+    runs-on: ubuntu-latest
+    needs: build
+    if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/DEVOPS-8-cloud' }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: cargil
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Build production app
+        run: docker build --target production --tag cargil/todo-app:${{ github.sha }} --tag cargil/todo-app:latest .
+
+      - name: Push production app to Docker Hub
+        run: docker push cargil/todo-app:${{ github.sha }} && docker push cargil/todo-app:latest
+
+      - name: Call azure webhook
+        run: curl -dH -X POST ${{ secrets.AZURE_WEBHOOK }}
```

Comment:
One nice addition here can be adding the [--fail](https://curl.se/docs/manpage.html#-f) flag, as otherwise curl reports success as long as a successful connection was made, even if the response reports a failure (e.g. a status code > 400) which can lead to confusing pipeline results